$NetBSD: patch-Makefile,v 1.2 2020/11/27 11:34:17 schmonz Exp $

Make sure openssl headers get found.
Not all systems have dprintf(); use ours for portability.

--- Makefile.orig	2016-09-21 22:05:29.000000000 +0000
+++ Makefile
@@ -5,8 +5,10 @@ TARBALL := ${DISTNAME}.tar.gz
 
 DEFINES += -D_XOPEN_SOURCE=700
 DEFINES += -D_BSD_SOURCE
+DEFINES += -Ddprintf=fdprintf
 CFLAGS_SSL=`pkg-config --cflags libssl`
-LIBS_TLS ?= -ltls `pkg-config --libs libssl`
+CFLAGS += $(CFLAGS_SSL)
+LIBS_TLS ?= -ltls `pkg-config --libs libssl openssl`
 LIBS_SSL = `pkg-config --libs libssl openssl`
 
 .PHONY: all test clean install
@@ -25,11 +27,11 @@ socks: socks.o
 httpc.o: http_parser.h
 http_parser.o: http_parser.h
 
-httpc: httpc.o http_parser.o
-	$(CC) -o $@ httpc.o http_parser.o
+httpc: httpc.o http_parser.o fdprintf.o
+	$(CC) -o $@ httpc.o http_parser.o fdprintf.o
 
-httppc: httppc.o http_parser.o
-	$(CC) -o $@ httppc.o http_parser.o
+httppc: httppc.o http_parser.o fdprintf.o
+	$(CC) -o $@ httppc.o http_parser.o fdprintf.o
 
 # SSL/TLS
 tlsc: tlsc.o
